app/scripts/lib/multichain-api/scope/scope.ts

```diff
@@ -16,16 +16,25 @@ export enum KnownCaipNamespace {
 
 export type Scope = CaipChainId | CaipReference;
 
-export type ScopeObject = {
+export type InternalScope = CaipChainId | KnownCaipNamespace.Wallet;
+
+export type ExternalScopeObject = InternalScopeObject & {
   scopes?: CaipChainId[];
+};
+
+export type InternalScopeObject = {
   methods: string[];
   notifications: string[];
   accounts?: CaipAccountId[];
   rpcDocuments?: string[];
   rpcEndpoints?: string[];
 };
 
-export type ScopesObject = Record<Scope, ScopeObject>;
+export type ExternalScopesObject = Record<Scope, ExternalScopeObject>;
+
+export type InternalScopesObject = Record<CaipChainId, InternalScopeObject> & {
+  [KnownCaipNamespace.Wallet]?: InternalScopeObject;
+};
```

Comment:
Could we add a few comments/tsdocs explaining these types?

Reply:
Can be pretty lightweight. Just explaining why there is a bifurcation between external and internal scopeObjects

Reply:
comments added here https://github.com/MetaMask/metamask-extension/pull/26690/commits/218ad5e1e3c74bc0b69366f09f35e9786a061396